types/relay-runtime/lib/util/getFragmentIdentifier.d.ts

```diff
@@ -0,0 +1,3 @@
+import { ReaderFragment } from './ReaderNode';
+
+export default function getFragmentIdentifier(fragmentNode: ReaderFragment, fragmentRef: any): string;
```

Comment:
Think there is a type for that fragmentRef: `FragmentRef`.

Reply:
@maraisr Which do you mean?

Reply:
@maraisr friendly ping

Reply:
Super sorry! This guy https://github.com/DefinitelyTyped/DefinitelyTyped/blob/903538301983042bd4ec105330e20e5a1a7f2a8a/types/relay-runtime/index.d.ts#L220

Reply:
This is used only by readInlineData
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/903538301983042bd4ec105330e20e5a1a7f2a8a/types/relay-runtime/lib/store/readInlineData.d.ts#L4
and it was done as well as relay:
https://github.com/facebook/relay/blob/master/packages/relay-runtime/store/readInlineData.js#L27

While in this case relay used the mixed type

Reply:
relay's Flow types have been known to not always be accurate. Granted im not entirely sure what "getFragmentIdentifier" does, but to me it feels like the argument should be more strict than "any".

Reply:
This is an internal function used in relay-experimental in order to calculate a unique fragment id.

I suppose that's why the fragmentRef typology has been relaxed.

on types/relay-runtime i would not deviate from their current implementation, at most you could propose a PR in relay and improve the typing and then report it here 👍